tests/baselines/reference/mappedTypeWithAny.js

```diff
@@ -51,10 +51,21 @@ declare function stringifyArray<T extends readonly any[]>(arr: T): { -readonly [
 let abc: any[] = stringifyArray(void 0 as any);
 
 declare function stringifyPair<T extends readonly [any, any]>(arr: T): { -readonly [K in keyof T]: string };
-let def: [any, any] = stringifyPair(void 0 as any);
+let def: [any, any] = stringifyPair(void 0 as any);
+
+// Repro from #46582
+
+export type Evolvable<E extends Evolver> = {
+  [P in keyof E]: never;
+};
+export type Evolver<T extends Evolvable<any> = any> = {
+  [key in keyof Partial<T>]: never;
+};
+

```

Comment:
Can you remove the export here, or make all of the types exported? This turned off `.d.ts` emit for the rest of the file.

Reply:
Done!